### PR TITLE
ENG-4944 Added "withdrawn" to indexed preprint titles

### PR DESCRIPTION
**Pull request by uditijmehta**
-   Ticket: [ENG-4944](https://openscience.atlassian.net/browse/ENG-4944)
-   Feature flag: n/a

## Purpose

Add "Withdrawn" to indexed preprint titles

## Summary of Changes

Added getter that checks if the preprint title is withdrawn, added title entry for withdrawn preprints in yml file

## Screenshot(s)

Screenshot 2024-03-07 at 12 32 51 PM (2)

## QA Notes

[ENG-4944]: https://openscience.atlassian.net/browse/ENG-4944?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by brianjgeiger** [comment]
Drive by: This is looking good, but I think we will also need to alter the title in the Head Tags and the title bar of the window for this to be complete. Probably the ticket should have called this out explicitly.

**Comment by brianjgeiger** [comment]
Also, well done on getting this to this point so quickly!

**Review by futa-ikeda — CHANGES_REQUESTED** [review]
Agree with what @brianjgeiger mentioned about updating some of the metadata we set on the page's head tag. This wasn't called out in the original ticket so I apologize for the bit of scope-creep here, but Google Scholar uses the metadata in the head tag to figure out titles. I believe there is a `setHeadTags` task on the `route.ts` file. If you could add some similar logic to update the title in there, that would be great. Very fast turn around on getting your first PR up as well! ⚡

File: app/preprints/detail/route.ts [hunk 1]
```diff
@@ -117,7 +117,7 @@ export default class PreprintsDetail extends Route {
             const image = 'engines-dist/registries/assets/img/osf-sharing.png';
 
             const metaTagsData = {
-                title: preprint.title,
+                title: preprint.isWithdrawn ? `Withdrawn: ${preprint.title}` : preprint.title,
```
Comment (brianjgeiger) [inline app/preprints/detail/route.ts:120]: Any reason not to use `this.intl.t` here?

File: package.json [hunk 2]
```diff
@@ -198,7 +198,7 @@
     "ember-sinon": "^5.0.0",
     "ember-sinon-qunit": "^6.0.0",
     "ember-sortable": "^5.0.0",
-    "ember-source": "~3.28.8",
+    "ember-source": "~3.28.12",
```
Comment (brianjgeiger) [inline package.json:201]: Is this the only thing that needs to change if we're bumping versions? Or are there other ember dependencies that also need to be updated?

Also, is this necessary for your PR, or would it make sense to do this as a separate PR so we can explore the above questions?

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
One suggestion to use the translated string in the `metaTagsData` and a couple of questions related to bumping the ember version.

File: yarn.lock [hunk 3]
```diff
@@ -12664,7 +12664,7 @@ ember-source-channel-url@^3.0.0:
   dependencies:
     node-fetch "^2.6.0"
 
-ember-source@~3.28.8:
+ember-source@~3.28.12:
```
Comment (brianjgeiger) [inline yarn.lock:12667]: Looks like you need to re-yarn as well.

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
It's verrry close. Need to re-yarn, and I think the logic on the title in the head tags is a little wrong. Other than that it looks great.

**Review by brianjgeiger — APPROVED** [review]
🎉 🐧 🎉

**Review by futa-ikeda — APPROVED** [review]
Woohoo! Great work!

**Pull request merged**